kalite/coachreports/api_views.py

```diff
@@ -139,24 +144,27 @@ def aggregate_learner_logs(request):
             latest_activity_timestamp__lte=end_date, **obj_ids).order_by("-latest_activity_timestamp")
 
 
-        if id_field == "video":
-            output_dict["content_time_spent"] += log_objects.aggregate(Sum("total_seconds_watched"))["total_seconds_watched__sum"]
-        elif id_field == "content":
-            output_dict["content_time_spent"] += log_objects.aggregate(Sum("time_spent"))["time_spent__sum"]
-        elif id_field == "exercise":
+        if log_type == "video":
+            output_dict["content_time_spent"] += log_objects.aggregate(Sum("total_seconds_watched"))["total_seconds_watched__sum"] or 0
+        elif log_type == "content":
+            output_dict["content_time_spent"] += log_objects.aggregate(Sum("time_spent"))["time_spent__sum"] or 0
+        elif log_type == "exercise":
             output_dict["exercise_attempts"] = AttemptLog.objects.filter(user__in=learners,
-                latest_activity_timestamp__gte=start_date,
-                latest_activity_timestamp__lte=end_date).count()
-            output_dict["exercise_mastery"] = log_objects.aggregate(Avg("streak_progress"))["streak_progress__avg"]
+                timestamp__gte=start_date,
+                timestamp__lte=end_date).count()
+            if log_objects.aggregate(Avg("streak_progress"))["streak_progress__avg"] is not None:
+                output_dict["exercise_mastery"] = round(log_objects.aggregate(Avg("streak_progress"))["streak_progress__avg"])
         output_logs.extend(log_objects)
 
+    # Report total time in hours
+    output_dict["content_time_spent"] = round(output_dict["content_time_spent"]/3600.0,1)
     output_logs.sort(key=lambda x: x.latest_activity_timestamp, reverse=True)
     output_dict["learner_events"] = [{
         "learner": log.user.get_name(),
         "complete": log.complete,
         "struggling": getattr(log, "struggling", None),
         "progress": getattr(log, "streak_progress", getattr(log, "progress", None)),
-        "content": get_exercise_cache().get(getattr(log, "exercise_id"), get_content_cache().get(getattr(log, "video_id", getattr(log, "content_id")), {})),
+        "content": get_exercise_cache().get(getattr(log, "exercise_id", ""), get_content_cache().get(getattr(log, "video_id", getattr(log, "content_id", "")), {})),
```

Comment:
Ensure there is a default value, so that log models without these fields don't cause errors.